spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/metadata/MongoDbMetadataStoreTests.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2002-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.integration.mongodb.metadata;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.integration.mongodb.rules.MongoDbAvailable;
+import org.springframework.integration.mongodb.rules.MongoDbAvailableTests;
+
+
+/**
+ * @author Senthil Arumugam, Samiraj Panneer Selvam
+ *
+ */
+public class MongoDbMetadataStoreTests extends MongoDbAvailableTests {
+	
+	private final static String DEFAULT_COLLECTION_NAME = "metadatastore";
+	private final String file1 = "/remotepath/filesTodownload/file-1.txt";
+	private final String file1Id = "12345";
+	private MongoDbMetadataStore store = null;
+	
+	@MongoDbAvailable
+	@Before
```

Comment:
No need for the `@MongoDbAvailable` on the `@Before`. 
From other side I don't see any reason in this  `@Before` method at all, since the logic of your test methods in many places are based on the fresh `MongoDbMetadataStore`